### PR TITLE
build: initialize const vector explicitly

### PR DESCRIPTION
**Pull request by muff1nman**
Description: Fix build on fedora by initializing const vector explicitly. 
Risk Level: Low
Testing: compile, unit/integration test
Docs Changes: N/A
Release Notes: N/A

### CODE REVIEW

File: source/common/http/async_client_impl.h [hunk 1]
```diff
@@ -144,9 +144,9 @@ class AsyncStreamImpl : public AsyncClient::Stream,
     }
     absl::optional<std::chrono::milliseconds> maxInterval() const override { return absl::nullopt; }
 
-    const std::vector<uint32_t> retriable_status_codes_;
-    const std::vector<Http::HeaderMatcherSharedPtr> retriable_headers_;
-    const std::vector<Http::HeaderMatcherSharedPtr> retriable_request_headers_;
+    const std::vector<uint32_t> retriable_status_codes_{};
```
Comment (yanavlasov) [inline source/common/http/async_client_impl.h]: It is a bit weird that this change is required here but not in a similar class NullHedgePolicy a few lines above. What toolset are you using to compile envoy?

Comment (mattklein123) [reply source/common/http/async_client_impl.h]: +1 this doesn't make much sense to me and seems like a compiler bug?

/wait

**Comment by lizan** [comment]
likely related: https://envoyproxy.slack.com/archives/C78HA81DH/p1573642623234200

https://gist.github.com/onemanbucket/8802b3cdfba06593bc3b21da590704ac

Could be a bug in libstdc++ though.

**Comment by jpeach** [comment]
Const member default initialization seems to be somewhere between "doesn't work" and "might work with some toolchains".

https://stackoverflow.com/questions/33809441/default-initialization-of-const-qualified-type-with-no-user-provided-constructor

godbolt doesn't like my simple example, though I can't tell which c++ standard library it is using

https://godbolt.org/z/NJhDqm

So IMHO, this fix is correct and useful.

**Comment by lizan** [comment]
@jpeach thanks for the explanation, makes sense to me.

**Comment by dnoe** [comment]
Can you merge master? I'm not sure what's up with the test failure.

**Comment by lizan** [comment]
Seems it's old AZP status hanged there. @mattklein123 or @htuch can you just force merge?

**Pull request merged**